### DISCUSSION TITLE
Add permission to change the username

### DISCUSSION THREAD

**Discussion by Anubarak** [craftcms/cms#9697] [Ideas]
Currently only [admins ](https://github.com/craftcms/cms/blob/develop/src/templates/users/_accountfields.html#L13) can change the `username`
```TWIG
{% if not craft.app.config.general.useEmailAsUsername %}
    {{ forms.textField({
        first: true,
        label: "Username"|t('app'),
        id: 'username',
        name: 'username',
        value: (user is defined ? user.username : null),
        autofocus: true,
        autocomplete: false,
        required: (isNewUser or (currentUser.admin or user.getIsCurrent()) ? true : false),
        disabled: (isNewUser or (currentUser.admin or user.getIsCurrent()) ? false : true),
        errors: (user is defined ? user.getErrors('username') : null),
        inputAttributes: {
            data: {lpignore: 'true'},
        },
    }) }}
{% endif %}
```

one of our clients asked for the permission to be able to change those as well and I noticed it isn't possible.
Could you implement a permission for this or is there a security reason behind that?

Edit: when you activate `useEmailAsUsername` non admin users can actually kinda change the `username` property indirectly so I'm not sure if there are really security concerns.

**Comment by brandonkelly** [comment]
Think we will just make this part of the existing “Administrate users” permission in Craft 4, which already covers changing users’ emails and passwords.

**Reply by Anubarak** [reply]
So no easy option to do that in Craft 3?

**Reply by brandonkelly** [reply]
3.7 was the last minor 3.x release, and I wouldn’t feel comfortable making a permission-related change in a patch release.
